diff --git a/cmdline.c b/cmdline.c
index 715b1d4..f45a879 100644
--- a/cmdline.c
+++ b/cmdline.c
@@ -425,6 +425,21 @@ int cmdline_process_param(CmdlineArg *arg, CmdlineArg *nextarg,
         }
     }
 
+#ifdef PUTTYNG
+    if (!stricmp(p, "-hwndparent")) {
+        RETURN(2);
+        hwnd_parent = (uintptr_t)strtoull(value, NULL, 0);
+        return 2;
+    }
+    if (!strcmp(p, "-auth-plugin") || !strcmp(p, "-auth_plugin")) {
+        RETURN(2);
+        UNAVAILABLE_IN(TOOLTYPE_NONNETWORK);
+        SAVEABLE(0);
+        conf_set_str(conf, CONF_auth_plugin, value);
+        return 2;
+    }
+#endif
+
     if (!strcmp(p, "-load")) {
         RETURN(2);
         /* This parameter must be processed immediately rather than being
